OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -78,11 +78,14 @@ public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
 
 		public bool CanEnterCell(World world, Actor self, CPos cell, Actor ignoreActor = null, bool checkTransientActors = true)
 		{
-			if (LocomotorInfo.MovementCostForCell(world, cell) == int.MaxValue)
+			var locomotor = world.WorldActor.TraitsImplementing<Locomotor>()
```

Comment:
This method is called from some reasonably hot paths, so doing this lookup is going to be painful.
Can we cache a `LocomotorInstance` (or something else that avoids the naming conflict) alongside or instead of `LocomotorInfo` from `RulesetLoaded`?